src/mongo/client/dbclientinterface.h

```diff
@@ -1080,7 +1081,7 @@ namespace mongo {
 
            See testDbEval() in dbclient.cpp for an example of usage.
         */
-        bool eval(const std::string &dbname, const std::string &jscode, BSONObj& info, BSONElement& retValue, BSONObj *args = 0);
+        bool eval(const std::string &dbname, const std::string &jscode, BSONObj& info, BSONElement& retValue, bool nolock=false, BSONObj *args = 0);
```

Comment:
I think that if you are adding a new default parameter, you must add it last. Otherwise callers passing a BSONObj for 'args' will suddenly get that coerced to bool and interpreted as a 'nolock' value, and still pass a NULL args, which probably isn't what they wanted.